### DISCUSSION TITLE
Multiple XML reports vs single XML report

### DISCUSSION THREAD

**Discussion by lostdog19** [Xray-App/xray-junit-extensions#39] [Q&A]
Is there a way to get multiple XML reports instead of just a single one?   For example, I've got several classes of tests, and It'd be very helpful for us to get 1 XML per class.  Something like this:

TEST-Class1.xml
TEST-Class2.xml
TEST-Class3.xml

**Comment by bitcoder** [comment]
Right now it generates just one file; it's non configurable.
To better understand your need, can you please elaborate why it would be helpful for you to have different files, one per each class?

**Reply by lostdog19** [reply]
There are a couple reasons.  

First, it appears that there are issues with XRAY when importing a larger set (more than 100) of test results when those tests are linked to requirements as part of the test metadata.

Second, as part of our automated process, we are being asked to create executions based on functionality groups as it aids in the reporting side of things. By splitting the file, that would allow us to easily do this.

**Reply by bitcoder** [reply]
I created a feature request: #43 ; we can track it there

**Reply by bitcoder** [reply]
Do you have opportunity to try out a beta @lostdog19 ?

**Reply by bitcoder** [reply]
@lostdog19  please try out the code in the branch https://github.com/Xray-App/xray-junit-extensions/tree/reports_per_class  and please let me know if it works as you expected

**Reply by bitcoder** [reply]
This is available as of v0.8.0
